source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -444,6 +446,8 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     }
     address_changed = true;
     stats_.host_address_changed_.inc();
+  } else if (current_address == nullptr) {
+    primary_host_info->host_info_->setDetails(details);
```

Comment:
this will end up setting error details for DNS even if we already have an address for the host, but this particular DNS query failed to give us an address (i.e. `current_address==nullptr`).  So probably instead of an `else if`, right after the `if` on L434, we should do `if (!primary_host_info->host_info_->hasAddress()) { primary_host_info->host_info_->setDetails(details); }`.

Unless I'm missing something?